src/data_device_manager/data_offer.rs

```diff
@@ -295,6 +311,31 @@ impl DataOfferData {
         }
     }
 
+    pub(crate) fn drop_performed(&self) {
+        let mut inner = self.inner.lock().unwrap();
+
+        if let DataDeviceOffer::Drag(ref mut o) = inner.deref_mut().offer {
+            o.dropped = true;
+        }
+    }
```

Comment:
Do we really need this as a separate function if it's used only once in `Drop` handler?